multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/cf/clients/CfRolesGetter.java

```diff
@@ -0,0 +1,51 @@
+package org.cloudfoundry.multiapps.controller.core.cf.clients;
+
+import com.sap.cloudfoundry.client.facade.CloudCredentials;
+import com.sap.cloudfoundry.client.facade.domain.UserRole;
+
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class CfRolesGetter extends CustomControllerClient {
+
+    private static final String ROLES_FILTER = Arrays.stream(UserRole.values())
+                                                     .map(UserRole::getName)
+                                                     .collect(Collectors.joining(","));
+    private static final String GET_ROLES_URL = "/v3/roles?space_guids=%s&user_guids=%s&types=" + ROLES_FILTER;
+
+    public CfRolesGetter(WebClientFactory webClientFactory, CloudCredentials credentials) {
+        super(webClientFactory, credentials);
+    }
+
+    public Set<UserRole> getRoles(UUID spaceGuid, UUID userGuid) {
+        return new CustomControllerClientErrorHandler().handleErrorsOrReturnResult(() -> doGetRoles(spaceGuid, userGuid));
+    }
+
+    private Set<UserRole> doGetRoles(UUID spaceGuid, UUID userGuid) {
+        String url = String.format(GET_ROLES_URL, spaceGuid, userGuid);
+        var list = getListOfResources(new UserRoleMapper(), url);
+        var result = EnumSet.noneOf(UserRole.class);
+        result.addAll(list);
+        return result;
+    }
```

Comment:
Use EnumSet.copyOf

Reply:
This will throw an exception if the collection is empty